expression/builtin_string.go

```diff
@@ -1364,6 +1364,11 @@ func (b *builtinSubstringIndexSig) evalString(row chunk.Row) (d string, isNull b
 	if len(delim) == 0 {
 		return "", false, nil
 	}
+	// when count > MaxInt64, returns whole string.
+	c, ok := b.args[2].(*Constant)
```

Comment:
In this case
```
select substring_index('aaa.bbb.ccc.ddd.eee.fff.ggg', '.', 18446744073709551615 - 1 + id)
from (select 1 as id) as t1
where id = 1;
```
the result is `ggg`, because of the type of `b.args[2]` is a plus function.

btw, don't forget`builtinSubstringIndexSig.vecEvalString`.

Reply:
Updated, PTAL. Thx

Reply:
The following case which using  scalar evaluate way still be failed.
```
	tk.Se.GetSessionVars().EnableVectorizedExpression = false
	tk.MustQuery(`select substring_index('aaa.bbb.ccc.ddd.eee', '.', 18446744073709551615 - 1 + id) from (select 1 as id) as t1`).Check(testkit.Rows(`aaa.bbb.ccc.ddd.eee`))
	tk.MustQuery(`select substring_index('aaa.bbb.ccc.ddd.eee', '.', -18446744073709551615 - 1 + id) from (select 1 as id) as t1`).Check(testkit.Rows(`aaa.bbb.ccc.ddd.eee`))
	tk.Se.GetSessionVars().EnableVectorizedExpression = true
```

Expression has two ways to evaluate value, vectorized and scalar.  You can use session var `session.GetSessionVars().EnableVectorizedExpression` to control to use which way.

Reply:
Should we use `mysql.HasUnsignedFlag(b.args[2].GetType().Flag)` like `vecEvalString`?

Reply:
Done, PTAL

Reply:
It's OK to me. PTAL @wshwsh12 @xhebox.